src/index.js

```diff
@@ -1,46 +1,17 @@
-import { spy, configure, getDebugName } from "mobx"
+import { observable, configure } from "mobx"
 import { Component } from "react"
 import { unstable_batchedUpdates as rdBatched } from "react-dom"
-import { unstable_batchedUpdates as rnBatched } from "react-native"
 
 if (!Component) throw new Error("mobx-react requires React to be available")
-if (!spy) throw new Error("mobx-react requires mobx to be available")
+if (!observable) throw new Error("mobx-react requires mobx to be available")
 
 if (typeof rdBatched === "function") configure({ reactionScheduler: rdBatched })
-else if (typeof rnBatched === "function") configure({ reactionScheduler: rnBatched })
 
-export {
-    observer,
-    Observer,
-    renderReporter,
-    componentByNodeRegistry as componentByNodeRegistery,
-    componentByNodeRegistry,
-    trackComponents,
-    useStaticRendering
-} from "./observer"
+export { useObservable, useComputed, useDisposable, useObserver, Observer } from "mobx-react-lite"
 
-export { default as Provider } from "./Provider"
-export { default as inject } from "./inject"
-export { disposeOnUnmount } from "./disposeOnUnmount"
-
-import * as propTypes from "./propTypes"
-export { propTypes }
-export { propTypes as PropTypes }
-
-import { errorsReporter } from "./observer"
-export const onError = fn => errorsReporter.on(fn)
+export { observer, useStaticRendering } from "./observer"
 
-/* DevTool support */
-// See: https://github.com/andykog/mobx-devtools/blob/d8976c24b8cb727ed59f9a0bc905a009df79e221/src/backend/installGlobalHook.js
-
-import { renderReporter, componentByNodeRegistry, trackComponents } from "./observer"
-if (typeof __MOBX_DEVTOOLS_GLOBAL_HOOK__ === "object") {
-    const mobx = { spy, extras: { getDebugName } }
-    const mobxReact = {
-        renderReporter,
-        componentByNodeRegistry,
-        componentByNodeRegistery: componentByNodeRegistry,
-        trackComponents
-    }
-    __MOBX_DEVTOOLS_GLOBAL_HOOK__.injectMobxReact(mobxReact, mobx)
-}
+export { Provider } from "./Provider"
```

Comment:
Note here we should probably export the Context object as well so people can make a custom hook for it. It's definitely better an easier migration than wanting from them to rewrite to new Context in a big swoop.